### PR TITLE
[UI] Receipts: Centralize import methods

### PR DESCRIPTION
**Pull request by lachlanjc**
## Summary of the problem

Before, the Receipt Center page had a bunch of grey text with instructions on how to import receipts. Copying the links required using context menus. 
CleanShot 2024-08-22 at 18 27 23@2x

For email, it sent you to your account settings to find or change your forwarding address, which then linked back to the Receipt Bin page (we refer to this page as Receipts, Receipt Center, Receipt Bin, and the URL is My Inbox).
CleanShot 2024-08-22 at 18 27 35@2x

## Describe your changes

I've centralized these instructions and configuration as parallel options at the top of the Receipts page:

CleanShot 2024-08-22 at 18 28 57@2x

### CODE REVIEW

File: app/views/events/g_suite_overview.html.erb
Comment (sampoder) [inline app/views/events/g_suite_overview.html.erb:1]: Fixed in https://github.com/hackclub/hcb/pull/7661 as well!

File: app/views/my/inbox.html.erb [hunk 1]
```diff
@@ -6,23 +6,55 @@
...
-  <p class="muted h3">
-    Alternatively, text your receipts to <%= sms_to "+1-864-548-4225", class: "break-none" %>
-    <% if @mailbox_address %>
-      or email them to <%= mail_to @mailbox_address.address, class: "break-none" %>.
-    <% else %>
-      or configure an email for receipt uploads in <%= link_to "your settings", my_settings_path %>.
-    <% end %>
-  </p>
+  <details class="my-4" <% @mailbox_address ? "open" : "" %>>
+    <summary>
```
Comment (sampoder) [inline app/views/my/inbox.html.erb]: Thoughts on always showing them?

Comment (lachlanjc) [reply app/views/my/inbox.html.erb]: It's not ideal to give up the vertical space at the top of the screen, but moving the section down doesn't make sense. If we drop the heading & card styling we won't use as much vertical space though!

File: app/views/mailbox_addresses/_mailbox_address.html.erb
Comment (garyhtou) [inline app/views/mailbox_addresses/_mailbox_address.html.erb:1]: BTW, the tooltips on the right of the card gets clipped
image

image

File: app/views/my/inbox.html.erb (hunk 1 above)
Comment (lachlanjc) [reply app/views/my/inbox.html.erb]: image

Comment (sampoder) [reply app/views/my/inbox.html.erb]: I personally prefer this style but don't mind either way, we can merge something in and then improve on top of that

**Review by sampoder — APPROVED** [review]
LGTM!

File: app/views/mailbox_addresses/_mailbox_address.html.erb
Comment (lachlanjc) [reply app/views/mailbox_addresses/_mailbox_address.html.erb:1]: It's true, but we've had that bug for 5 years…

Comment (garyhtou) [reply app/views/mailbox_addresses/_mailbox_address.html.erb:1]: No longer relevant since the card was dropped!

**Review by garyhtou — APPROVED** [review]
Good to go!

**Comment by garyhtou** [comment]
Thank you for these PRs @lachlanjc!

**Pull request merged**